src/Components/Activities/index.js

```diff
@@ -1,11 +1,79 @@
+import { useEffect, useState } from 'react';
 import styles from './activities.module.css';
+import Table from './Table';
+import Form from './Form';
+
+const Activities = () => {
+  const [activities, setActivities] = useState([]);
+  const [showForm, setShowForm] = useState(false);
+  const [selectedActivity, setSelectedActivity] = useState(null);
+
+  const getActivities = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/activity/`);
+      const responseData = await response.json();
+      const data = responseData.data;
+      setActivities(data);
+    } catch (error) {
+      alert('Error fetching activities: ' + error);
+    }
+  };
+
+  useEffect(() => {
+    getActivities();
+  }, []);
+
+  const addActivity = ({ name, description }) => {
+    const newActivity = {
+      _id: Math.floor(Math.random() * 100),
+      name,
+      description
+    };
+    setActivities([...activities, newActivity]);
+    setShowForm(false);
+  };
+
+  const editActivity = (updatedActivity) => {
```

Comment:
same as with create, we are only updating the data that we have at the moment we need to impact the DB with the changes